### DISCUSSION TITLE
Update type for a model does not have nor id nor parentId

### DISCUSSION THREAD

**Discussion by PixelsCommander** [blitz-js/blitz#1392] [Q&A]
Prisma`s Update type for a model does not have nor id nor parentId. 

The situation is - there is an item that has a category.

While editing an item I get the whole Item into its form values. When I try to submit a form update mutation errors on backend saying that id and categoryId are not expected to be on the ItemUpdate type.

The first one is solvable by removing id from the form data and keeping it into where only. This is annoying but works. But without categoryId I can not actually update category. Is this a bug? What do I do wrong?

**Comment by PixelsCommander** [comment]
Seem I had to read Prisma basics again. The behavior is odd but expected

**Reply by flybayer** [reply]
So you got it figured out?

**Reply by PixelsCommander** [reply]
Still not sure. It looks very odd and adds complexity to a very simple case. Let me explain it with code so maybe you can confirm or say what I am doing wrong.

```
export default async function updateEvent({ where, data }: UpdateEventInput, ctx: Ctx) {
  // @ts-ignore
  let categoryId = parseInt(data.categoryId)
  if (categoryId) {
    // @ts-ignore
    data.categoryId = undefined
    data.brand = {
      connect: { id: categoryId },
    }
  }

  const event = await db.event.update({ where, data })
```

As you see categoryId do not exist on update type so I had even to ts-ignore it. While in reality setting this field would be the most straightforward way of doing the thing when having one - many relations. While the connect construction I had to use blows mutation for no reason except the missing field on the type.

If prisma keeps categoryId field in EventUpdateArg it would be just:

```
export default async function updateEvent({ where, data }: UpdateEventInput, ctx: Ctx) {
  const event = await db.event.update({ where, data })
```

But right now it is TS Error: categoryId property does not exist on EventUpdateArg

**Reply by flybayer** [reply]
Ok, yeah this is a bit of a friction point with prisma.

What I do is add `categoryId` to the input type (to fix the TS errors), and then basically do like you have with `connect: {id: categoryId}`. Once you get this in your head, it's not bad. But yeah hopefully it can be improved at some point.

So your mutations will almost always be more complex than just

```
export default async function updateEvent({ where, data }: UpdateEventInput, ctx: Ctx) {
  const event = await db.event.update({ where, data })
```
